salt/modules/pkg_resource.py

```diff
@@ -311,21 +311,30 @@ def format_pkg_list(packages, versions_as_list, attr):
     '''
     ret = copy.deepcopy(packages)
     if attr:
+        ret_attr = {}
         requested_attr = {'epoch', 'version', 'release', 'arch', 'install_date', 'install_date_time_t'}
 
         if attr != 'all':
-            requested_attr &= set(attr + ['version'])
+            requested_attr &= set(attr + ['version'] + ['arch'])
 
         for name in ret:
+            _parse_arch_from_name = __salt__.get('pkg.parse_arch_from_name', lambda pkgname: {'name': pkgname, 'arch': None})
```

Comment:
Let's not try to get too cute, here. What about something like this?

```python
if 'pkg.parse_arch_from_name' in __salt__:
    _parse_arch_from_name = __salt__['pkg.parse_arch_from_name'](pkgname)
else:
    _parse_arch_from_name = {'name': pkgname, 'arch': None}
```

Also, I have a _strong_ dislike of the function name `parse_arch_from_name`. We should think of something better.

Reply:
I agree with @terminalmage here on both points. We should not be using lambdas for something like this. As for the method name, a simpler `parse_arch` seems fine to me.

Reply:
Hey @dwoz , your feedback is now addressed. No more lambda and function has been renamed to `parse_arch`.